lib/matplotlib/tests/test_backend_bases.py

```diff
@@ -157,3 +159,26 @@ def test_interactive_zoom():
 
     tb.zoom()
     assert ax.get_navigate_mode() is None
+
+
+def test_toolbar_zoompan():
+    plt.rcParams['toolbar'] = 'toolmanager'
+    ax = plt.gca()
+    assert ax.get_navigate_mode() is None
+    with pytest.warns(UserWarning) as rec:
+        ax.figure.canvas.manager.toolmanager.add_tool(name="zoom",
+                                                      tool=ToolZoom)
+        ax.figure.canvas.manager.toolmanager.add_tool(name="pan",
+                                                      tool=ToolPan)
+        ax.figure.canvas.manager.toolmanager.add_tool(name=_views_positions,
+                                                      tool=ToolViewsPositions)
+        ax.figure.canvas.manager.toolmanager.add_tool(name='rubberband',
+                                                      tool=RubberbandBase)
+        ax.figure.canvas.manager.toolmanager.trigger_tool('zoom')
+        assert ax.get_navigate_mode() == "ZOOM"
+        ax.figure.canvas.manager.toolmanager.trigger_tool('pan')
+        assert ax.get_navigate_mode() == "PAN"
+        assert(len(rec) == 4)
+        for r in rec:
+            assert("The new Tool classes introduced in v1.5 are experimental"
+                   in str(r.message))
```

Comment:
I'm not sure this needs to be asserted as part of _this_ test specifically. However, if this is to ensure that other _unexpected_ warnings are not raised, then it can be done by specifying the `match` argument in the `pytest.warns` call.

Reply:
@QuLogic do you want this to be resolved before merging?

Reply:
I was hoping for a reply from @xordux about the intention at least.

Reply:
Oops I thought it's an advice that `pytest.warns` can also be used for the same. @QuLogic Yes, while asserting warning messages my intentions were to ensure that any other _unexpected_ warnings are not raised.

Reply:
I think we should switch to `pytest.warns` then.

Reply:
@QuLogic I have added the `match` argument in `pytest.warns` but now some other test cases are failing which don't seem to be related to my changes.